ui/app/orders/controllers/orderFulfillmentController.js

```diff
@@ -52,12 +52,43 @@ angular.module('bahmni.orders').controller('OrderFulfillmentController', ['$scop
                 });
             });
         };
+
+        var getObservationsForOrders = function () {
+            var orderLabelConcept = appService.getAppDescriptor().getConfigValue("orderLabelConcept");
```

Comment:
This could be logic in backend (seems to be domain logic in frontend)

Reply:
This method is to fetches the concept from the backend to determine the radiologist assigned to the order, group the data by orderUuid, and set the label value accordingly. Could you please provide more details on what you mean by 'domain logic in frontend'

Reply:
I was thinking, that an endpoint in the backend could perform this logic, and just give you data in a structure where you get the assigned radiologist/provider and value from backend as-is, and UI just inspects the response and renders (so, the logic moves to the backend). If you feel this is not going to work, and there isn't a backend API that can handle this logic, then you can ignore this comment.

Reply:
@angshu

Reply:
To provide further context, based on my and @kavitha-sundararajan  assessment, there currently isn't an existing API that aligns with Gurpreet's proposed approach. The fetch API utilized in this instance is common within bahmniapps and is not tailored to the specific requirements Gurpreet has suggested. Therefore, it appears that developing a new API or modifying an existing one would be necessary to accommodate this

Reply:
After careful discussion, we have decided to proceed with the changes. @umair-fayaz and @kavitha-sundararajan explained the necessity of this decision, and our verification confirmed that the changes do not cause any disruptions. As a future enhancement, we can consider adding a new endpoint to simplify the process or potentially replace this backend call with existing alternatives.

cc: @angshu, @gsluthra, @gokultw, @binduak